### PR TITLE
install: add backtrace on failure

### PR DESCRIPTION
**Pull request by iMichka**
The error message might be useless alone

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
This was suggested by @claui. (Thanks again it helped me debug my issue).

I can put the printing beyond the debug flag, but I think printing it out right away is what we want here.

**Comment by claui** [comment]
> I can put the printing beyond the debug flag, but I think printing it out right away is what we want here.

No strong opinion on that one because we’re already applying it inconsistently.

Examples:

- [always print](https://github.com/Homebrew/brew/blob/db998860f11e0ad8b6d9dc6570a639007688a4ad/Library/Homebrew/brew.rb#L180)
- [print only if we’re debugging](https://github.com/Homebrew/brew/blob/db998860f11e0ad8b6d9dc6570a639007688a4ad/Library/Homebrew/brew.rb#L160)

File: Library/Homebrew/cmd/install.rb [hunk 1]
```diff
@@ -267,6 +267,7 @@ def install
     # Need to rescue before `FormulaUnavailableError` (superclass of this)
     # is handled, as searching for a formula doesn't make sense here (the
     # formula was found, but there's a problem with its implementation).
+    puts e.backtrace
```
Comment (Bo98) [inline Library/Homebrew/cmd/install.rb:270]: I'm going to be picky and suggest:

```suggestion
    $stderr.puts e.backtrace
```

Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/install.rb:270]:
```suggestion
    $stderr.puts e.backtrace if Homebrew::EnvConfig.developer?
```

**Comment by MikeMcQuaid** [comment]
> No strong opinion on that one because we’re already applying it inconsistently.

The rule (or at least, what the rule should be) is pretty much is "do we expect users to report this issue to us".

**Comment by iMichka** [comment]
Agreed with @MikeMcQuaid's version: it's mostly useful for us as devs when something fails while refactoring code or trying stuff out.

**Comment by MikeMcQuaid** [comment]
Thanks again @iMichka!

**Pull request merged**